test/metabase/upload_test.clj

```diff
@@ -472,21 +472,47 @@
           (when grant?
             (perms/revoke-data-perms! group-id db-id)))))))
 
+(defn do-with-uploads-allowed
+  "Set uploads-enabled to true, and uses an admin user, run the thunk"
+  [thunk]
+  (mt/with-temporary-setting-values [uploads-enabled true]
+    (mt/with-current-user (mt/user->id :crowberto)
+      (thunk))))
+
+(defmacro with-uploads-allowed [& body]
+  `(do-with-uploads-allowed (fn [] ~@body)))
+
+(defn- do-with-upload-table! [t f]
+  (try (f t)
+       (finally
+         (let [table-ident (#'upload/table-identifier t)]
+           (driver/drop-table! driver/*driver* (mt/id) table-ident)))))
+
+(defmacro with-upload-table!
+  "Execute `body` with a table created by evaluating the `table-creator` expression. The table instance is bound to
+  `table-sym` in `body`. The table is cleaned up from both the test and app DB after the body executes.
+
+    (with-upload-table [table (create-upload-table! ...)]
+      ...)"
+  {:style/indent 1}
+  [[table-sym table-creator] & body]
```

Comment:
```suggestion
  [[table-binding create-table-fn] & body]
```

- table-binding: I think this is how we name these kind of pseudo binding in macro
- create-table-fn: when I first read `table-creator` I thought it's an user, not a function that create a table

Reply:
oh I just realized `table-creator` is not a function but a table, in that case I think we can call it `upload-table` since as far as the macro concerned it's evaluated as a table.

Reply:
Good point, `table-creator` is a it misleading. I went with `create-table-expr` to make it clear that the table needs to be created inside the expression and not before it, which is the main sharp edge when using this macro.